viite-UI/src/model/ProjectCollection.js

```diff
@@ -230,6 +231,21 @@
     };
 
     this.saveProjectLinks = function(changedLinks, statusCode) {
+      var thisone = this;
+      var validUserGivenAddrMValues = function(linkId, userEndAddr){
+        if(_.isUndefined(userEndAddr))
+          return false;
+          var roadPartIds = self.getMultiSelectIds(linkId);
+          var roadPartLinks = self.getByLinkId(roadPartIds);
+          var sortedGeometries = _.map(_.sortBy((roadPartLinks), function(link){return link.getData().endAddressM;}), function(roadPart){
```

Comment:
I think you don't need to sort the roadPartLinks, just sum the length of each ones.